src/Database/DBStructure.php

```diff
@@ -176,14 +176,14 @@ public static function dryRun(): string
 	public static function performUpdate(bool $enable_maintenance_mode = true, bool $verbose = false): string
 	{
 		if ($enable_maintenance_mode) {
-			DI::keyValue()->set('maintenance', 1);
+			DI::config()->set('system', 'maintenance', 1);
```

Comment:
Why did you revert to config?

Reply:
because the maintenance mode is necessary at the very beginning of the load and for me it's context is more part of the config, not of a keyvalue cache